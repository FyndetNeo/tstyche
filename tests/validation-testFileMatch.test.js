import { afterEach, describe, expect, test } from "@jest/globals";
import { clearFixture, getFixtureUrl, writeFixture } from "./__utils__/fixtureFactory.js";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const isStringTestText = `import { expect, test } from "tstyche";
test("is string?", () => {
  expect<string>().type.toBeString();
});
`;

const fixtureUrl = getFixtureUrl("validation-testFileMatch", { generated: true });

afterEach(async () => {
  await clearFixture(fixtureUrl);
});

describe("'testFileMatch' configuration file option", () => {
  test("when option argument is not a list", async () => {
    const config = {
      testFileMatch: "feature",
    };

    await writeFixture(fixtureUrl, {
      ["__typetests__/dummy.test.ts"]: isStringTestText,
      ["tstyche.config.json"]: JSON.stringify(config, null, 2),
    });

    const { exitCode, stderr, stdout } = await spawnTyche(fixtureUrl);

    expect(stdout).toBe("");
    expect(normalizeOutput(stderr)).toMatchSnapshot("stderr");

    expect(exitCode).toBe(1);
  });

  test("when item of the list is not a string", async () => {
    const config = {
      testFileMatch: ["examples/*", false],
    };

    await writeFixture(fixtureUrl, {
      ["__typetests__/dummy.test.ts"]: isStringTestText,
      ["tstyche.config.json"]: JSON.stringify(config, null, 2),
    });

    const { exitCode, stderr, stdout } = await spawnTyche(fixtureUrl);

    expect(stdout).toBe("");
    expect(normalizeOutput(stderr)).toMatchSnapshot("stderr");

    expect(exitCode).toBe(1);
  });
});
