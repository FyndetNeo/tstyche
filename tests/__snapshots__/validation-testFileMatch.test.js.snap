// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'testFileMatch' configuration file option when item of the list is not a string: stderr 1`] = `
"Error: Item of the 'testFileMatch' list must be of type string.

  2 |   "testFileMatch": [
  3 |     "examples/*",
> 4 |     false
    |     ^
  5 |   ]
  6 | }

      at ./tstyche.config.json:4:5

"
`;

exports[`'testFileMatch' configuration file option when option argument is not a list: stderr 1`] = `
"Error: Option 'testFileMatch' requires an argument of type list.

  1 | {
> 2 |   "testFileMatch": "feature"
    |                    ^
  3 | }

      at ./tstyche.config.json:2:20

"
`;
